### PR TITLE
[WIP] Rename the Profiler to Inspector

### LINKED ISSUE #34042: [RFC] Renaming some core concepts... for the better
I'd like to rename two concepts (with deprecation notices in 4.4 and full switch in 5.0): *Symfony Environments* and the *Symfony
Profiler*.

I know, they are core concepts, but I think it's time to reconsider their names. The impact will be huge in the docs, and our current users will probably hate us, but I think this is for the best going forward.

**Symfony environments, why is it confusing?**

We now manipulate *environment* variables everywhere in Symfony.
Most cloud providers also have *environments*.

You are testing the prod environment (the Symfony one) for your development
environment (local machine). See what I mean?

A Symfony environment is a way to switch the configuration to a different set of values: from dev to prod or the other way around. Or test.

I'd like to rename it to **Symfony mode**.

`APP_ENV=prod` to `APP_MODE=prod`

You are testing the prod mode on your development environment.

**What about the Symfony Profiler?**

I made the original mistake. A profiler is something very specific, like
XHProf, or Blackfire.

The Symfony profiler is not a profiler in the usual sense of the word. That's annoying and I keep bumping into people that do not understand the difference with Blackfire for instance.

There is also an inconsistency as we have a web **debug** toolbar, not a web
profiler toolbar. Another confusion, why is the name different?

I'd like to rename it to **Symfony Inspector** (was **Symfony Debugger** initially, but I changed my mind -- see comments below). I think it describes what it does best and also convey the notion that it should not be deployed in production. It is also the name used by browsers, so it matches well with our user expectations.

I know that this might seems pointless, but naming things correctly helps
developers understand the concepts.

WDYT?

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | yes 
| Tickets       | Fix #34042 
| License       | MIT
| Doc PR        | -

This PR is proudly sponsored by the Blackfire Profiler :) https://blackfire.io/

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/Bundle/TestBundle/Resources/config/routing.yml [hunk 1]
```diff
@@ -22,9 +22,9 @@ session_showflash:
     path:     /session_showflash
     defaults: { _controller: Symfony\Bundle\FrameworkBundle\Tests\Functional\Bundle\TestBundle\Controller\SessionController::showFlashAction }
 
-profiler:
-    path:     /profiler
-    defaults: { _controller: Symfony\Bundle\FrameworkBundle\Tests\Functional\Bundle\TestBundle\Controller\ProfilerController::indexAction }
+inspector:
+    path:     /inspector
+    defaults: { _controller: Symfony\Bundle\FrameworkBundle\Tests\Functional\Bundle\TestBundle\Controller\InspectorController::indexAction }
```
Comment (lolautruche) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/Bundle/TestBundle/Resources/config/routing.yml]: Shouldn’t we add a redirection from `/profiler` to `/inspector` for 4.4?
I know this is only for dev mode, but some people might consider this as a BC break.

**Review by lolautruche — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -217,18 +217,18 @@ private function addFragmentsSection(ArrayNodeDefinition $rootNode)
         ;
     }
 
-    private function addProfilerSection(ArrayNodeDefinition $rootNode)
+    private function addInspectorSection(ArrayNodeDefinition $rootNode)
     {
         $rootNode
             ->children()
                 ->arrayNode('profiler')
```
Comment (xuanquynh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: This configuration key should be renamed to inspector, shouldn't it?

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 3]
```diff
@@ -327,7 +327,7 @@ public function load(array $configs, ContainerBuilder $container)
         $this->registerSsiConfiguration($config['ssi'], $container, $loader);
         $this->registerFragmentsConfiguration($config['fragments'], $container, $loader);
         $this->registerTranslatorConfiguration($config['translator'], $container, $loader, $config['default_locale']);
-        $this->registerProfilerConfiguration($config['profiler'], $container, $loader);
+        $this->registerInspectorConfiguration($config['profiler'], $container, $loader);
```
Comment (xuanquynh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: This depends on the above configuration key.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php (hunk 2 above)
Comment (lolautruche) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: That would be a big BC break.

Comment (stof) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: well, we can have a BC layer for that. That's the big benefit of the semantic configuration system.

**Review by stof — COMMENTED** [review]
what about WebProfilerBundle and the `Profiler` namespace and class name in HttpKernel ? This renaming is still partial.

**Pull request closed without merging**